tpm/internal/socket/socket.go

```diff
@@ -0,0 +1,23 @@
+package socket
+
+import (
+	"errors"
+	"io"
+)
+
+var (
+	ErrNotAvailable = errors.New("socket not available")
+	ErrNotSupported = errors.New("connecting to a TPM using a UNIX socket is not supported on Windows")
```

Comment:
Sounds good.

There is limited support for UNIX sockets on [Windows](https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/), but the proper solution are named pipes using https://github.com/microsoft/go-winio, for example, `step` connects to the agent using:

```golang
conn, err := winio.DialPipeContext(context.Background(), `\\.\\pipe\\openssh-ssh-agent`)
```

The same library can create a listener on a pipe using:
```golang
lis, err := winio.ListenPipe(`\\.\\pipe\\step-agent`, nil) 
```

If the simulator in windows can use those, they will not be difficult to implement in a future PR.